spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/sampler/UuidToBigIntegerConverterTest.java

```diff
@@ -0,0 +1,52 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import lombok.Data;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.math.BigInteger;
+import java.util.UUID;
+
+import static java.math.BigInteger.ONE;
+import static java.math.BigInteger.ZERO;
+import static junitparams.JUnitParamsRunner.$;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.springframework.cloud.sleuth.sampler.UuidToBigIntegerConverterTest.TestData.sample;
+
+@RunWith(JUnitParamsRunner.class)
+public class UuidToBigIntegerConverterTest {
+
+	private static final BigInteger TWO = new BigInteger("2");
+
+	@Test
+	@Parameters
+	public void should_convert_uuid_to_big_integer(TestData testData) throws Exception {
+		UUID uuid = new UUID(testData.mostSignificantBits, testData.leastSignificantBits);
+
+		BigInteger bigInteger = UuidToBigIntegerConverter.uuidToBigInt(uuid);
+
+		then(bigInteger).isEqualByComparingTo(testData.expectedInteger);
+	}
+
+	public Object[] parametersForShould_convert_uuid_to_big_integer() {
+		return $(
+				sample(Long.MAX_VALUE, Long.MAX_VALUE, TWO.pow(126).subtract(ONE)),
```

Comment:
Heh the length of that explanation implies we need a simpler input. One
shouldn't need to figure so many things out to know if the inputs are right
or not.

Food for thought:

If the goal is to prove we can cover all 128 bits, how about initialize the
longs as 0xFFFF.... Then on the RHS, use the array ctor of BigInteger with
an array filled by Arrays.fill((byte)0xFF) or more explicitly an array
literal made of 0xFFs

Even if you alternate one can visually compare left with right without any
math. This means less likely one will just assume the inputs and outputs
are correct out of laziness. This could help prevent bugs from sticking
around.

Remember, at the end of the day, the bigint is just a convenient holder of
128 random bits for the purpose of comparing them as a number.